SpatialGDK/Build/Scripts/ExternalSchemaCodegen.bat

```diff
@@ -0,0 +1,70 @@
+@echo off
+
+IF NOT "%~2"=="" IF "%~3"=="" GOTO START
+ECHO This script requires two parameters (both defined relative to project root):
+ECHO - path to external schema directory
+ECHO - target output folder for generated code
+exit /b 1
+
+:START
+
+call :MarkStartOfBlock "Setup variables"
+  set GDK_FOLDER=%~dp0\..\..\..
+  set GAME_FOLDER=%GDK_FOLDER%\..\..\..
+  set SCHEMA_COMPILER_PATH=%GDK_FOLDER%\SpatialGDK\Binaries\ThirdParty\Improbable\Programs\schema_compiler.exe
+  set CODEGEN_EXE_PATH=%GDK_FOLDER%\SpatialGDK\Binaries\ThirdParty\Improbable\Programs\CodeGenerator.exe
+  set SCHEMA_STD_COPY_DIR=%GAME_FOLDER%\spatial\build\dependencies\schema\standard_library
+	set BUNDLE_CACHE_DIR=%GDK_FOLDER%\SpatialGDK\Intermediate\ExternalSchemaCodegen
+	set SCHEMA_BUNDLE_FILE_NAME=external_schema_bundle.json
+call :MarkEndOfBlock "Setup variables"
+
+call :MarkStartOfBlock "Clean folders"
+    rd /s /q "%BUNDLE_CACHE_DIR%"           2>nul
+call :MarkEndOfBlock "Clean folders"
+
+call :MarkStartOfBlock "Create folders"
+    md "%BUNDLE_CACHE_DIR%"            >nul 2>nul

```

Comment:
Should this create the output folder for generated code also?

Reply:
This is done by the code generator tool when it tries to output, this means you don't get the minor false positive of a created output folder when the script can still fail on an intermediary step like the codegen. This was my reasoning, happy to move here if you prefer?

Reply:
No that's reasonable.